dev/io.openliberty.security.jakartasec.4.0.internal/src/io/openliberty/security/jakartasec40/identitystore/InMemoryIdentityStoreDefinitionWrapper.java

```diff
@@ -0,0 +1,443 @@
+                    Tr.debug(tc, "evaluatePriority", "Returning null since priorityExpression is a deferred expression and this is called on initialization.");
+                }
+                return null;
+            }
+
+            Tr.warning(tc, "JAKARTASEC_WARNING_IDSTORE_CONFIG", new Object[] { "priority/priorityExpression", JakartaSec40Constants.SPEC_DEFAULT_PRIORITY });
+            return JakartaSec40Constants.SPEC_DEFAULT_PRIORITY;
+        }
+    }
+
+    /**
+     * Evaluate and return the useFor.
+     *
+     * @param immediateOnly If true, only return a non-null value if the setting is either an
+     *                          immediate EL expression or not set by an EL expression. If false, return the
+     *                          value regardless of where it is evaluated.
+     * @return The useFor or null if immediateOnly==true AND the value is not evaluated
+     *         from a deferred EL expression.
+     */
+    @FFDCIgnore(IllegalArgumentException.class)
+    private Set<ValidationType> evaluateUseFor(boolean immediateOnly) {
+
+        String useForExpression = inMemoryIdentityStoreDefinition.useForExpression();
+        ValidationType[] useFor = inMemoryIdentityStoreDefinition.useFor();
+        try {
+            return elHelper.processUseFor(useForExpression, useFor, immediateOnly);
+        } catch (PropertyNotFoundException e) {
+            Tr.warning(tc, "JAKARTASEC_WARNING_IDSTORE_CONFIG", new Object[] { "useFor/useForExpression", JakartaSec40Constants.SPEC_DEFAULT_USEFOR });
+            return Set.of(JakartaSec40Constants.SPEC_DEFAULT_USEFOR);
+        } catch (IllegalArgumentException e) {
+            /*
+             * If deferred expression and called during initialization, return null so the expression can be re-evaluated
+             * again later.
+             */
+            if (immediateOnly && ELHelperWrapper.isDeferredExpression(useForExpression)) {
+                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                    Tr.debug(tc, "evaluateUseFor", "Returning null since useForExpression is a deferred expression and this is called on initialization.");
+                }
+                return null;
+            }
+
+            Tr.warning(tc, "JAKARTASEC_WARNING_IDSTORE_CONFIG", new Object[] { "useFor/useForExpression", JakartaSec40Constants.SPEC_DEFAULT_USEFOR });
+
+            return Set.of(JakartaSec40Constants.SPEC_DEFAULT_USEFOR);
+        }
+    }
+
+    /**
+     * Get the priority for the {@link IdentityStore}.
+     *
+     * @return The priority.
+     *
+     * @see InMemoryIdentityStoreDefinition#priority()
+     * @see InMemoryIdentityStoreDefinition#priorityExpression()
+     */
+    int getPriority() {
+        if (this.priority != null) {
+            return this.priority;
+        }
+        Integer priority = null;
+        return (priority = evaluatePriority(false)) != null ? priority : JakartaSec40Constants.SPEC_DEFAULT_PRIORITY;
```

Comment:
```suggestion
        Integer evalPriority;
        return (evalPriority = evaluatePriority(false)) != null ? evalPriority : JakartaSec40Constants.SPEC_DEFAULT_PRIORITY;
```
Using the same variable name as the class scoped variable is confusing.

Reply:
Yes, agreed.

Done.